Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m

```diff
@@ -141,54 +122,37 @@ - (UIView *)mediaView {
             }
         }
     } else if ([self isAudio]) {
-        NSError *err = NULL;
-        NSURL *url =
-            [MIMETypeUtil simLinkCorrectExtensionOfFile:_attachment.mediaURL ofMIMEType:_attachment.contentType];
-
-        if (!self.waveform) {
-            AVURLAsset *asset         = [[AVURLAsset alloc] initWithURL:url options:nil];
-            self.waveform                 = [[SCWaveformView alloc] init];
-            self.waveform.frame           = CGRectMake(42.0, 0.0, size.width - 84, size.height);
-            self.waveform.asset           = asset;
-            self.waveform.progressColor   = [UIColor whiteColor];
-            self.waveform.backgroundColor = [UIColor colorWithRed:229 / 255.0f green:228 / 255.0f blue:234 / 255.0f alpha:1.0f];
-            [self.waveform generateWaveforms];
-            self.waveform.progress = 0.0;
-        }
-
         UIView *audioBubble = [[UIView alloc] initWithFrame:CGRectMake(0.0, 0.0, size.width, size.height)];
         audioBubble.backgroundColor =
             [UIColor colorWithRed:10 / 255.0f green:130 / 255.0f blue:253 / 255.0f alpha:1.0f];
         audioBubble.layer.cornerRadius = 18;
         audioBubble.layer.masksToBounds = YES;
 
-        _audioPlayPauseButton = [[UIButton alloc] initWithFrame:CGRectMake(3, 3, 30, 30)];
+        _audioPlayPauseButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, kAudioViewWidth, kAudioButtonHeight)];
         _audioPlayPauseButton.enabled = NO;
 
-        AVAudioPlayer *player = [[AVAudioPlayer alloc] initWithContentsOfURL:url error:&err];
-        _durationLabel        = [[UILabel alloc] init];
-        _durationLabel.text   = [self formatDuration:player.duration];
-        _durationLabel.font   = [UIFont systemFontOfSize:14];
-        [_durationLabel sizeToFit];
-        _durationLabel.frame = CGRectMake((size.width - _durationLabel.frame.size.width) - 10,
-                                          _durationLabel.frame.origin.y,
-                                          _durationLabel.frame.size.width,
-                                          AUDIO_BAR_HEIGHT);
-        _durationLabel.backgroundColor = [UIColor clearColor];
-        _durationLabel.textColor       = [UIColor whiteColor];
+        NSString *audioLabelText = [[MIMETypeUtil fileExtensionForMIMEType:self.contentType] uppercaseString];
```

Comment:
nit: on the attachment confirmation page we print "Mp3" not "MP3" (probably the confirmation page is what's wrong, and this should stay the same)